sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/tests/TimeSeriesInsightsQuerySeriesTests.cs

```diff
@@ -0,0 +1,217 @@
+                        .And
+                         .OnlyContain(timeStamp => timeStamp <= endTime);
+                        seriesEventsPage.Properties.Count.Should().Be(2); // Temperature and Humidity
+                        seriesEventsPage.Properties.Should().Contain((property) => property.Name == Temperature)
+                        .And
+                         .Contain((property) => property.Name == Humidity);
+                    }
+
+                    return null;
+                }, MaxNumberOfRetries, s_retryDelay);
+
+                // Send 2 events with a special condition that can be used later to query on
+                IDictionary<string, object> messageBase = QueryTestsHelper.BuildMessageBase(modelSettings.TimeSeriesIdProperties.ToArray(), tsiId);
+                messageBase[QueryTestsHelper.Temperature] = 1.2;
+                messageBase[QueryTestsHelper.Humidity] = 3.4;
+                string messageBody = JsonSerializer.Serialize(messageBase);
+                var message = new Message(Encoding.ASCII.GetBytes(messageBody))
+                {
+                    ContentType = "application/json",
+                    ContentEncoding = "utf-8",
+                };
+
+                Func<Task> sendEventAct = async () => await deviceClient.SendEventAsync(message).ConfigureAwait(false);
+                sendEventAct.Should().NotThrow();
+
+                // Send it again
+                sendEventAct.Should().NotThrow();
+
+                // Query for the two events with a filter
+                querySeriesRequestOptions.Filter = "$event.Temperature.Double = 1.2";
+                await TestRetryHelper.RetryAsync<AsyncPageable<QueryResultPage>>(async () =>
+                {
+                    AsyncPageable<QueryResultPage> querySeriesEventsPages = tsiClient.QuerySeriesAsync(tsiId, startTime, endTime, querySeriesRequestOptions);
+                    await foreach (QueryResultPage seriesEventsPage in querySeriesEventsPages)
+                    {
+                        seriesEventsPage.Timestamps.Should().HaveCount(2);
+                        seriesEventsPage.Properties.Should().HaveCount(2)
+                        .And
+                         .Contain((property) => property.Name == Temperature)
+                        .And
+                         .Contain((property) => property.Name == Humidity);
+
+                        var temperatureValues = seriesEventsPage
+                        .Properties
+                        .First((property) => property.Name == Temperature)
+                        .Values.Cast<double>().ToList();
+                        temperatureValues.Should().AllBeEquivalentTo(1.2);
+                    }
+
+                    return null;
+                }, MaxNumberOfRetries, s_retryDelay);
+
+                // Query for the two events with a filter, but only take 1
```

Comment:
Not a strong opinion but I can also see these can be different tests or even methods ( I can see why you chose this approach of single test probably because device client creation and sending event can be done all at once)

1. Query all, 
2. Query specific events, 
3. Query with filter .. etc

Reply:
Yup. I also felt it was getting a little too much for a single test method. But the reason you mentioned is exactly why I have 1 test for the different scenarios. 

But I like the idea of separating them. I will create a task to break down the test after we make good progress on all the other priorities.